messaging/src/main/java/org/axonframework/eventhandling/deadletter/EvaluationTask.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.deadletter;
+
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventMessageHandler;
+import org.axonframework.eventhandling.ListenerInvocationErrorHandler;
+import org.axonframework.messaging.deadletter.DeadLetter;
+import org.axonframework.messaging.deadletter.DeadLetterEvaluationException;
+import org.axonframework.messaging.deadletter.DeadLetterQueue;
+import org.axonframework.messaging.unitofwork.DefaultUnitOfWork;
+import org.axonframework.messaging.unitofwork.UnitOfWork;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A {@link Runnable} implementation used to evaluate a {@link DeadLetter} taken from the {@link DeadLetterQueue}. This
+ * task is added through {@link DeadLetterQueue#onAvailable(String, Runnable)}, so we can typically assume there are
+ * entries ready for evaluation.
+ *
+ * @author Steven van Beelen
+ * @since 4.6.0
+ */
+class EvaluationTask implements Runnable {
```

Comment:
Is there a clearer name we can think of than `EvaluationTask`?

Reply:
`RetryAvailableDeadLetterTask`?

Reply:
Small adjustment proposal of @gklijs : `RetryAvailableDeadLettersTask` (or: `DeadLetterQueueRetryTask`

Reply:
Renamed to `DeadLetterEventProcessingTask`, as the API on the Dead Letter Queue no refers to "processing" instead of "release" or "retry."